src/data/finset.lean

```diff
@@ -171,6 +171,15 @@ theorem singleton_inj {a b : α} : ι a = ι b ↔ a = b :=
 
 @[simp] lemma coe_singleton (a : α) : ↑(ι a) = ({a} : set α) := rfl
 
+lemma singleton_iff_unique_mem (s : finset α) : (∃ a, s = finset.singleton a) ↔ ∃! a, a ∈ s :=
```

Comment:
This might be better stated in a slightly stronger way without the exists.
```lean
lemma eq_singleton_iff {s : finset α} {a : α} : 
  s = finset.singleton a ↔ a ∈ s ∧ ∀ x ∈ s, x = a
```

That way we know both the `a`s are the same, and the stated theorem should be a trivial consequence of this, `by simp only [eq_singleton_iff_unique_mem, exists_unique]` does the trick.